synthesis/fragment/fragment.go

```diff
@@ -178,5 +190,13 @@ func optimizeOverhangIteration(sequence string, minFragmentSize int, maxFragment
 // last 4 base pairs are the initial overhang set.
 func Fragment(sequence string, minFragmentSize int, maxFragmentSize int, existingOverhangs []string) ([]string, float64, error) {
 	sequence = strings.ToUpper(sequence)
-	return optimizeOverhangIteration(sequence, minFragmentSize, maxFragmentSize, []string{}, append([]string{sequence[:4], sequence[len(sequence)-4:]}, existingOverhangs...))
+	return optimizeOverhangIteration(sequence, minFragmentSize, maxFragmentSize, []string{}, append([]string{sequence[:4], sequence[len(sequence)-4:]}, existingOverhangs...), []string{})
+}
+
+// FragmentWithOverhangs fragments a sequence with only a certain overhang set.
+// This is useful if you are constraining the set of possible overhangs when
+// doing more advanced forms of cloning.
+func FragmentWithOverhangs(sequence string, minFragmentSize int, maxFragmentSize int, existingOverhangs []string, buildOverhangs []string) ([]string, float64, error) {
```

Comment:
I think instead of existing overhangs and build overhangs it should be `excludeOverhangs` and `includeOverhangs`.

Reply:
Those names make things clearer for me too